steps/generate_dataset_from_htme.py

```diff
@@ -429,6 +431,14 @@ def log_end_of_batch(correlation_id, run_id, status, dynamodb=None):
                          correlation_id, run_id, str(ex))
         sys.exit(-1)
 
+def create_adg_status_csv(correlation_id, s3_prefix, publish_bucket, s3_client):
```

Comment:
We would also need some unit tests for this. For reference I have written some tests in the branch attached to this ticket.